src/Drupal/Driver/Cores/Drupal7.php

```diff
@@ -324,13 +324,7 @@ public function termCreate(\stdClass $term) {
 
     \taxonomy_term_save($term);
 
-    // Loading a term by name returns an array of term objects, but there should
-    // only be one matching term in a testing context, so take the first match
-    // by reset()'ing $matches.
-    $matches = \taxonomy_get_term_by_name($term->name);
-    $saved_term = reset($matches);
-
-    return $saved_term;
+    return $term;
```

Comment:
@jhedstrom yeah, since $term is an object it gets the tid after save. The same happens with menu_link_save(), node_save(), etc. I've tested it with a fairly complex test suite and it worked well.

You can even see in the taxonomy_term_save function it uses the tid from the $term object to insert into the hierarchy table (https://api.drupal.org/api/drupal/modules!taxonomy!taxonomy.module/function/taxonomy_term_save/7)